### PR TITLE
2635 enhance UNet doc for the typical use case

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2635 .

### Description
This PR added the explanation from @ericspod to the doc-string of UNet.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/nets/unet.py [hunk 1]
```diff
@@ -64,18 +64,21 @@ def __init__(
         Note: The acceptable spatial size of input data depends on the parameters of the network,
             to set appropriate spatial size, please check the tutorial for more details:
             https://github.com/Project-MONAI/tutorials/blob/master/modules/UNet_input_size_constrains.ipynb.
-            Typically, applying `resize`, `pad` or `crop` transforms can help adjust the spatial size of input data.
+            Typically, when using a stride of 2 in down / up sampling, the output dimensions are either half of the
+            input when downsampling, and twice when upsampling, so if have N numbers of layers in the network,
```
Comment (ericspod) [inline monai/networks/nets/unet.py:68]: I would change this line to "input when downsampling, or twice when upsampling. In this case with N numbers of layers in the network"

**Review by ericspod — APPROVED** [review]

**Comment by ericspod** [comment]
Minor change but otherwise this should help explain things a bit more.

File: monai/networks/nets/unet.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/networks/nets/unet.py]: Sure, updated in the latest commit.
Thanks for your review!

**Pull request merged**